### ISSUE TITLE
Move exercises to subdirectory

### ISSUE THREAD

**Issue by kytrinyx** [exercism/xecmascript#187]
The problems api (x-api) now supports having exercises collected in a subdirectory
named `exercises`.

That is to say that instead of having a mix of `bin`, `docs`, and individual exercises,
we can have `bin`, `docs`, and `exercises` in the root of the repository, and all
the exercises collected in a subdirectory.

In other words, instead of this:

```
x{TRACK_ID}/
├── LICENSE
├── README.md
├── bin
│   └── fetch-configlet
├── bowling
│   ├── bowling_test.ext
│   └── example.ext
├── clock
│   ├── clock_test.ext
│   └── example.ext
├── config.json
└── docs
│   ├── ABOUT.md
│   └── img
... etc
```

we can have something like this:

```
x{TRACK_ID}/
├── LICENSE
├── README.md
├── bin
│   └── fetch-configlet
├── config.json
├── docs
│   ├── ABOUT.md
│   └── img
├── exercises
│   ├── bowling
│   │   ├── bowling_test.ext
│   │   └── example.ext
│   └── clock
│       ├── clock_test.ext
│       └── example.ext
... etc
```

This has already been deployed to production, so it's safe to make this change whenever you have time.

**Comment by matthewmorgan** [comment]
@rchavarria I'd be happy to work on this if you like.

Perhaps you could check out my open PR #186 and we could merge that first?  If it looks good I can make the same modifications to the JS track.

**Comment by rchavarria** [comment]
Yep, it would be better to merge #186. Please, merge it if you want and you can start working on this one. It would be awesome. Root folder will be much cleaner.

**Comment by matthewmorgan** [comment]
Closed by #188
